index.html

```diff
@@ -83,95 +69,172 @@
   });
-
-  function setup() {
-    // SVG
-
-    svg = d3.select('#chart')
+ function Chart() {
+     var chart = this;
+ 
+     chart.svg = d3.select('#chart1')
       .append('svg')
       .attr('width', width + margin.right + margin.left)
       .attr('height', height + margin.top + margin.bottom)
       .append('g')
       .attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
-
     // SCALES
-
-    x = d3.scale.linear()
-      .domain([0, d3.max(data, function (d) { return d.unemployment; })])
+    chart.x = d3.scale.linear()
+      .domain(d3.extent(data, function (a) {return a.year; }))
       .range([0, width])
       .nice();
-
-    y = d3.scale.linear()
-      .domain(d3.extent(data, function (d) { return d.inflation; }))
+    chart.y = d3.scale.linear()
+      .domain(d3.extent(data, function (a) {return a.value; }))
       .range([height, 0])
       .nice();
-
+    
     // AXES
-
-    // EVENT HANDLERS
-
-    d3.select('#year-range').on('change', function () {
-      options.yearRange = d3.event.target.value;
-      update();
+    // format x axis
+    var formatyears = d3.format("1000")
+    // format y axis
+    var formatmoney = d3.format("$1,000,000,000")
+    //Define X axis
+   
+    var xAxis = d3.svg.axis()
+             .scale(chart.x)
+             .orient("bottom")
+             .ticks(20)
+             .tickSize(height + 15)
+             .tickFormat(formatyears);
+    //Define Y axis
+    var yAxis = d3.svg.axis()
+             .scale(chart.y)
+             .tickSize(width)
+             .tickFormat(formatmoney)
+             .orient("right")
+             .ticks(5);
+   
+    //Create X axis
+     gx = this.svg.append("g")
+          .attr("class", "axis")
+          .attr("transform", "translate(0," + height + ")")
+          .append("text")
+          .attr("x", width)
+          .attr("y", 0)
+          .style("text-anchor", "end")
+          .text("Year");
+    //Create y axis
+      gy = this.svg.append("g")
+      .attr("transform", "rotate(-90)")
+      .attr("class", "axis")
+      .append("text")
+      .style("text-anchor", "end")
+      .text("Incentive Value") 
+      .attr("x", -10)
+      .attr("y", -10);
+      gy = this.svg.append("g")
+      .attr("class", "y axis")
+      .call(yAxis);
+    gy.selectAll("g").filter(function(d) { return d; })
+      .classed("minor", true);
+    
+      gx = this.svg.append("g")
+      .attr("class", "x axis")
+      .call(xAxis);
+      
+    gx.select('g').filter(function(d) { return d; })
+        .classed("minor", true);
+    gy.selectAll("text")
+    .attr("x", 4)
+    .attr("dy", -4);
+      // EVENT HANDLERS
+    d3.select('#location').on('change', function () {
+      options.location = d3.event.target.value;
+
+
+      chart.tooltip = d3.tip()
+         .attr('class', 'tooltip')
+         .offset([-6, 0])
+         .html(function(d) { return d.length });
+ 
+     chart.tooltip(chart.svg);
+ 
+     chart.update();
     });
```

Comment:
So chart.update() is running with the event handler. This means that chart.update() is only running when some changes the button with id #location.

This is why no circles are appearing the first time the code runs. You want chart.update() both within the event handler, and right after the instantiation of chart (I made a comment there as well).